knowledge-base/adr/0020-generator-options.md

```diff
@@ -37,26 +37,35 @@ We should discuss the details of those defaults and make a concious decision on
 | **New options**                                          |
-| -                                                        | `--[no]-strictNaming`            | Duplicate names are renamed by default.                                                           | Duplicate names throw an error by default. Disable validation to rename duplicates.                                                                          |
-| -                                                        | `--verbose`                      | Logs everything that happens during generation by deafult.                                        | Log only success / error per service by default. Enable verbosity through this flag.                                                                         |
-| -                                                        | `--version`, `-v`                | -                                                                                                 | Prints the version of the generator.                                                                                                                         |
+| -                                                        | `--[no]-strictNaming`            | Duplicate names are renamed by default.                                                           | Duplicate names throw an error by default. Disable validation to rename duplicates.                                                                                                                            |
+| -                                                        | `--verbose`                      | Logs everything that happens during generation by deafult.                                        | Log only success / error per service by default. Enable verbosity through this flag.                                                                                                                           |
+| -                                                        | `--version`, `-v`                | -                                                                                                 | Prints the version of the generator.                                                                                                                                                                           |
+
+\* Service Mapping - In OData we currently generate a `service-mapping.json` by default.
+Some services require a configuration to exist, in order to specify a service path, that could not be parsed from the specification by default.
+In those cases we plan to fail the generation when no service mapping was provided, similar to the behavior of `strictNaming`.
```

Comment:
DO we want to fail. I think warnig is finde with a remark to the `withCustomServicePath`  or config file option. At runtime if no path is given we have to fail also giving the options to include the file.

Reply:
Adjusted it. I would suggest, that the final behavior in that situation should be discussed as well and noted down.